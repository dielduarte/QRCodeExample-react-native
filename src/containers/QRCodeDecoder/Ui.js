import React, { Fragment } from 'react';
import QRdecoder from 'react-native-qrimage-decoder';

import Button from '../../components/Button';

const Ui = ({ source, showImagePicker, onSuccess, onError }) => {
  return (
    <Fragment>
      <Button title="Scan from gallery" onPress={showImagePicker} />
      <QRdecoder src={source} onSuccess={onSuccess} onError={onError} />
    </Fragment>
  );
};

export default Ui;
