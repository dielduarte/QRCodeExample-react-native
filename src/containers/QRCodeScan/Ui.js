import React, { Fragment } from 'react';

import Button from '../../components/Button';
import Camera from '../../components/Camera';

const Ui = ({ showCamera, setShowCamera, onBarCodeRead }) => {
  return (
    <Fragment>
      <Button title="Scan" onPress={() => setShowCamera(!showCamera)} />
      {showCamera && <Camera onBarCodeRead={onBarCodeRead} />}
    </Fragment>
  );
};

export default Ui;
