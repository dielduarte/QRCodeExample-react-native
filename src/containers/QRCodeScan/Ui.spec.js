import React from 'react';
import { shallow } from 'enzyme';

import Ui from './Ui';

describe('QRCodeScan Ui', () => {
  it('should render correctly', () => {
    const props = { showCamera: true, setShowCamera: () => {}, onBarCodeRead: () => {} };
    const Component = shallow(<Ui props={props} />);
    expect(Component).toMatchSnapshot();
  });
});
