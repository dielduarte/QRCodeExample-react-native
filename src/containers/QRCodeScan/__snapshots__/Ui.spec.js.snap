// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRCodeScan Ui should render correctly 1`] = `
<Fragment>
  <Button
    onPress={[Function]}
    title="Scan"
  />
</Fragment>
`;
